azure-toolkit-libs/azure-toolkit-auth-lib/src/main/java/com/microsoft/azure/tools/auth/core/maven/AzureCredential.java

```diff
@@ -0,0 +1,122 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for
+ * license information.
+ */
+
+package com.microsoft.azure.tools.auth.core.maven;
+
+import com.microsoft.aad.adal4j.AuthenticationResult;
+import com.microsoft.aad.adal4j.UserInfo;
+
+public class AzureCredential {
+
+    private String accessTokenType;
+    private String idToken;
+    private UserInfo userInfo;
+    private String accessToken;
+    private String refreshToken;
+    private boolean isMultipleResourceRefreshToken;
+    private String defaultSubscription;
+    private String environment;
+
+    /***
+     * Create an AzureCredential with {@link com.microsoft.aad.adal4j.AuthenticationResult}.
+     *
+     * @param result the AuthenticationResult, must not be null
+     * @return the newly created AzureCredential
+     */
+    public static AzureCredential fromAuthenticationResult(AuthenticationResult result) {
+        if (result == null) {
+            throw new IllegalArgumentException("Parameter \"result\" cannot be null");
+        }
+        final AzureCredential token = new AzureCredential();
+        token.setAccessTokenType(result.getAccessTokenType());
+        token.setAccessToken(result.getAccessToken());
+        token.setRefreshToken(result.getRefreshToken());
+        token.setIdToken(result.getIdToken());
+        token.setUserInfo(result.getUserInfo());
+        token.setMultipleResourceRefreshToken(result.isMultipleResourceRefreshToken());
+        return token;
+    }
+
+    public String getAccessTokenType() {
+        return accessTokenType;
+    }
+
+    public void setAccessTokenType(String accessTokenType) {
+        this.accessTokenType = accessTokenType;
+    }
+
+    public String getIdToken() {
```

Comment:
using lombok?

Reply:
it is from original auth-helper, will be deleted soon.